src/Adapter/Cart/CartPresenter.php

```diff
@@ -229,6 +229,50 @@ public function addCustomizedData(array $products, Cart $cart)
         }, $products);
     }
 
+    protected function determineRealShippingStatusAndText($cart)
+    {
+        $configuration = Configuration::getMultiple(array(
+            'PS_SHIPPING_FREE_PRICE',
+            'PS_SHIPPING_FREE_WEIGHT'
+        ));
+
+        // Check if is real free shiping based on current cart
+        $shippingCost = $cart->getTotalShippingCost(null, $this->includeTaxes());
+        $orderTotalwithDiscounts = $cart->getOrderTotal(true, Cart::BOTH_WITHOUT_SHIPPING, null, null, false);
+        $freeShippingFromPrice = (float) $configuration['PS_SHIPPING_FREE_PRICE'] > 0 && $orderTotalwithDiscounts >= (float) $configuration['PS_SHIPPING_FREE_PRICE']
+                                ? true
+                                : false
+        ;
```

Comment:
I find this a bit hard to read. Usually ternary to return `true` or `false` are useless (even if there are many in the codebase).

This should be enough I think:
```php
$freeShippingFromPrice = (float) $configuration['PS_SHIPPING_FREE_PRICE'] > 0
                                             && $orderTotalwithDiscounts >= (float) $configuration['PS_SHIPPING_FREE_PRICE'];
```